x-pack/legacy/plugins/epm/server/lib/elasticsearch/template/install.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { safeLoad } from 'js-yaml';
+import { RegistryPackage, DataSet, AssetReference } from '../../../../common/types';
+import * as Registry from '../../../registry';
+import { CallESAsCurrentUser } from '../../../../server/lib/cluster_access';
+import { getAssetsData } from '../../../packages/assets';
+import { Field } from '../../fields/field';
+import { generateMappings, getTemplate, generateTemplateName } from './template';
+
+const isFields = (path: string) => {
+  return path.includes('/fields/');
+};
+
+/**
+ * installTemplates installs one template for each dataset
+ *
+ * For each dataset, the fields.yml files are extracted. If there are multiple
+ * in one datasets, they are merged together into 1 and then converted to a template
+ * The template is currently loaded with the pkgey-package-dataset
+ * @param callCluster
+ * @param pkgkey
+ */
+export async function installTemplates(p: RegistryPackage, callCluster: CallESAsCurrentUser) {
```

Comment:
Would `PackageInfo` work here? If so, I'd prefer to use that.

So far, we've used `RegistryPackage` exclusively for typing the responses from the Registry API and hiding the shape from any code not in `server/registry`. This allows us to keep a stable interface for EPM code and prevent spreading Registry details.

Reply:
Yes, that should work too. As far as I can see PackageInfo is a superset of RegistryPackage  so it contains all the needed information. The reason I ended up with RegistgryPackage is because it is returned by `Registry.fetchInfo`. I see there is `getPackageInfoByKey` but it is in public. What you recommend to be used  to gather this info?